ethcore/src/migrations/account_bloom.rs

```diff
@@ -67,14 +79,32 @@ pub fn upgrade_account_bloom(db_path: &Path) -> Result<(), Error> {
 	};
 	let state_root = HeaderView::new(&best_block_header).state_root();
 
-	if check_bloom_exists(&source) {
-		// bloom already exists, nothing to do
-		trace!(target: "migration", "Bloom already present, skipping");
-		return Ok(())
+	let db = ::std::sync::Arc::new(source);
+	let batch = DBTransaction::new(&db);
+
+	if check_bloom_exists(&db) {
+		match check_space_match(&db) {
+			Ok(_) => {
+				// bloom already exists and desired and stored spaces match, nothing to do
+				trace!(target: "migration", "Bloom already present of the right space, skipping");
+				return Ok(())
+			},
+			Err(wrong_size) => {
+				// nullify existing bloom entries
+				trace!(target: "migration", "Clearing old bloom of space {}", &wrong_size);
+				let mut key = [0u8; 8];
+				for idx in 0..(wrong_size as u64/8) {
+					LittleEndian::write_u64(&mut key, idx);
+					try!(batch.put(DB_COL_ACCOUNT_BLOOM, &key, &[0u8; 8]));
+				}
+
+				LittleEndian::write_u64(&mut key, ACCOUNT_BLOOM_SPACE as u64);
+				try!(batch.put(DB_COL_ACCOUNT_BLOOM, ACCOUNT_BLOOM_SPACE_KEY, &key));
+			},
+		}
 	}
 
-	println!("Adding accounts bloom (one-time upgrade)");
-	let db = ::std::sync::Arc::new(source);
+	println!("Adding/expanding accounts bloom (one-time upgrade)");
```

Comment:
mix of printing and logging in this function. logging should be favored over printing wherever possible